### PR TITLE
Add Role cell to hosts listing

### PR DESCRIPTION
**Pull request by jtomasek**

### CODE REVIEW

File: frontend/public/metalkube/components/host/role-cell.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import React, { Fragment } from 'react';
+
+const getHostRole = machine =>
+  _.get(
+    machine,
+    ['metadata', 'labels', 'machine.openshift.io/cluster-api-machine-role'],
+    '-'
+  );
+
+const RoleCell = ({ machine }) => <Fragment>{getHostRole(machine)}</Fragment>;
```
Comment (suomiy) [inline frontend/public/metalkube/components/host/role-cell.jsx]: there is no need for the fragment

File: frontend/public/metalkube/components/host/role-cell.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import * as _ from 'lodash-es';
+import React, { Fragment } from 'react';
+
+const getHostRole = machine =>
+  _.get(
```
Comment (suomiy) [inline frontend/public/metalkube/components/host/role-cell.jsx]: this should probably go to some selectors

File: frontend/public/metalkube/components/host/host.tsx [hunk 3]
```diff
@@ -55,7 +62,21 @@ const HostRow = ({ obj: host }) => {
     },
   } = host;
 
-  const machineName = _.get(host, 'status.machineRef.name');
+  const machineName = _.get(host, 'spec.machineRef.name');
```
Comment (jtomasek) [inline frontend/public/metalkube/components/host/host.tsx]: I'll replace this with a selector call once https://github.com/kubevirt/web-ui-components/pull/333 merges

Comment (jtomasek) [reply frontend/public/metalkube/components/host/host.tsx]: Done

File: frontend/public/metalkube/components/host/host.tsx [hunk 4]
```diff
@@ -55,7 +62,21 @@ const HostRow = ({ obj: host }) => {
     },
   } = host;
 
-  const machineName = _.get(host, 'status.machineRef.name');
+  const machineName = getHostMachineName(host);
+  const machineResource = {
```
Comment (rawagner) [inline frontend/public/metalkube/components/host/host.tsx]: you can use `getResource` from `web-ui-components` which has a lot of these fields prefilled

Comment (jtomasek) [reply frontend/public/metalkube/components/host/host.tsx]: yeah, I tried but problem is that `getResource` only outputs resource kind as `Model.kind` and this does not work for machines as machines are stored in redux store by key which is a result of `referenceForModel(MachineModel)`. So I had to specify the machineResource explicitly here.

I think the general goal is to keep all resources keyed using regerenceForModel in redux, so we'll have to update getResource function to handle it.

**Pull request merged**